wagtail/documents/forms.py

```diff
@@ -60,7 +60,22 @@ def save(self, commit=True):
         return self.instance
 
     class Meta:
-        widgets = {"tags": AdminTagWidget, "file": forms.FileInput()}
+        widgets = {
+            "tags": AdminTagWidget,
+            "file": forms.FileInput(
+                attrs={
+                    "data-controller": "w-sync",
+                    "data-w-sync-target-value": "#id_title",
```

Comment:
We also need to pass the event name (`wagtail:documents-upload`) as a Stimulus value from here (and do the same for images)

```suggestion
                    "data-w-sync-target-value": "#id_title",
                    "data-w-sync-event-value": "wagtail:documents-upload"
```